odk2-src/build-scripts.rst

```diff
@@ -0,0 +1,221 @@
+.. spelling::
+  androidlibrary
+  androidcommon
+  gradle
+  Gradle
+  prebuilt
+  configs
+  Artifactory
+  Jacoco
+
+Build Scripts
+====================
+
+.. _build-scripts-prerequisites:
+
+Prerequisites
+------------------
+
+The Android tools, like most Android projects, use `Gradle <https://gradle.org/>`_ for compilation scripting. Before reading this document, you should be familiar with `Gradle for Android <https://developer.android.com/studio/build/index.html>`_.
+
+
+.. _build-scripts-directory-structure:
+
+Directory Structure
+------------------------
+
+The build scripts for the Android tools expect a particular directory structure. They expect a parent directory that contains each of them at the same level. This is **optional**.
+
+If you had all of the Android tools checked out your directory structure would look like this::
+
+  /opendatakit/
+      /androidlibrary/
+      /androidcommon/
+      /app-designer/
+      /gradle-config/
+      /scan/
+      /sensorsframework/
+      /sensorsinterface/
+      /services/
+      /suitcase/
+      /survey/
+      /sync-client/
+      /tables/
+
+There are two cases where this directory structure makes a difference:
+
+  - **Library Projects**:
+
+    - If androidlibrary and androidcommon are present in the same directory, according to the above structure, as the Android tools, then they will build against the local copy. If you want to make changes to Services and androidlibrary simultaneously, for example, this structure would be necessary.
+
+    - If the library projects are not present in the above configuration then a prebuilt binary will be downloaded according to the flavor you are building. For example, new binaries are posted on Snapshot for each commit, or on Master for each release.
+
+  - **Gradle Config**: If the gradle-config project is present in the above configuration, the gradle files in that folder will be used. Otherwise the release version specifies in :file:`settings.gradle` will be used.
+
+.. _build-scripts-building:
+
+
+Building the Android Tools
+-------------------------------
+
```

Comment:
Add a discussion about that we have two flavor dimensions. Describe each dimension's purpose and possible values.